crates/context/interface/src/context.rs

```diff
@@ -23,6 +23,8 @@ pub trait ContextTr {
     type Journal: JournalTr<Database = Self::Db>;
     /// Chain type
     type Chain;
+    /// Local context type
+    type Local: LocalContextTr;
```

Comment:
Local context is added to support hashing of initcode bytes and to cache validation result when they are called in TXCREATE.

LocalContext will later contain buffer from SharedMemory, so it makes sense to add it.